### PR TITLE
:sparkles: total of all dices / and sub of 2

### PR DESCRIPTION
**Pull request by fabienjuif**

### CODE REVIEW

File: src/app.jsx [hunk 1]
```diff
@@ -42,16 +44,35 @@ const App = () => {
...
+            {dices.reduce((acc, { roll = 0 }) => acc + roll, 0)}
+          </span>
+        </div>
+      )}
+
+      {dices.length === 2 && (
+        <div>
+          <span>Subtract</span>
+          <span>
+            {Math.abs(dices[0].roll - dices[1].roll) || ''}
```
Comment (delphinemillet) [inline src/app.jsx]: the default value should be a number

File: src/app.jsx [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
   const [printModal, setPrintModal] = useState(false)
 
-  const [dices, setDices] = useLocalStorage('dices', [])
-  const reset = () => setDices([])
-  const roll = () => dispatch('@@ui/ROLL>DICES')
+  const { dices, reset, add } = useContext(DicesContext)
 
   const addDice = (faces) => {
-    setDices([...dices, { id: Date.now(), faces }])
+    add({
```
Comment (delphinemillet) [inline src/app.jsx]: I don't understand the purpose of the handler `addDice`.
why the hook cannot handle the `id` setting ?

Comment (fabienjuif) [reply src/app.jsx]: 1. `addDice` helps me 
  a. add a dice to the context
  b. close the modal

2. I can move the id into `dicesContext.add`, in which case it should returns the added dice

Comment (delphinemillet) [reply src/app.jsx]: I missed the close modal part ^^
I prefer this way where the `dicesContext.add` takes care of everything related to dice modifications

**Review by delphinemillet — APPROVED** [review]

**Pull request merged**